packages/frontend/@n8n/design-system/src/v2/components/Select/Select.vue

```diff
@@ -0,0 +1,361 @@
+<script
+	setup
+	lang="ts"
+	generic="
+		T extends ArrayOrNested<SelectItem>,
+		VK extends GetItemKeys<T> = 'value',
+		M extends boolean = false
+	"
+>
+import { reactivePick } from '@vueuse/core';
+import {
+	SelectItem as RSelectItem,
+	SelectValue as RSelectValue,
+	SelectContent,
+	SelectGroup,
+	SelectItemIndicator,
+	SelectItemText,
+	SelectLabel,
+	SelectPortal,
+	SelectRoot,
+	SelectScrollDownButton,
+	SelectScrollUpButton,
+	SelectSeparator,
+	SelectTrigger,
+	SelectViewport,
+	useForwardPropsEmits,
+} from 'reka-ui';
+import { computed, useCssModule, useTemplateRef } from 'vue';
+
+import Icon from '@n8n/design-system/components/N8nIcon/Icon.vue';
+import { get, isArrayOfArray } from '@n8n/design-system/v2/utils';
+import type {
+	ArrayOrNested,
+	GetItemKeys,
+	GetModelValue,
+	NestedItem,
+} from '@n8n/design-system/v2/utils/types';
+
+import type {
+	SelectEmits,
+	SelectItem,
+	SelectProps,
+	SelectSlots,
+	SelectValue,
+} from './Select.types';
+
+defineOptions({ inheritAttrs: false });
+
+const $style = useCssModule();
+
+const props = withDefaults(defineProps<SelectProps<T, VK, M>>(), {
+	valueKey: 'value' as never,
+	labelKey: 'label' as never,
+	placeholder: 'Select an option',
+	selectedIcon: 'check',
+	variant: 'default',
+	size: 'small',
+});
+const emit = defineEmits<SelectEmits<T, VK, M>>();
+const slots = defineSlots<SelectSlots<T, VK, M>>();
+
+const rootProps = useForwardPropsEmits(
+	reactivePick(props, 'open', 'defaultOpen', 'disabled', 'autocomplete', 'required', 'multiple'),
+	emit,
+);
+
+const groups = computed<SelectItem[][]>(() => {
+	if (!props.items?.length) return [];
+	return isArrayOfArray(props.items) ? props.items : [props.items];
+});
+
+const triggerRef = useTemplateRef<InstanceType<typeof SelectTrigger>>('trigger');
+
+function isSelectItem(item: SelectItem): item is Exclude<SelectItem, SelectValue> {
+	return typeof item === 'object' && item !== null;
+}
+
+function castToSelectItemValue(
+	value?: GetModelValue<T, VK, M>,
+): Exclude<SelectItem, boolean> | Array<Exclude<SelectItem, boolean>> | undefined {
+	return value as Exclude<SelectItem, boolean> | Array<Exclude<SelectItem, boolean>> | undefined;
+}
+
+defineExpose({
+	triggerRef,
+});
+
+const variants = {
+	default: $style.Default,
+	ghost: $style.Ghost,
+};
+
+const variant = computed(() => variants[props.variant]);
+
+const sizes = {
```

Comment:
I'd suggest adding a constraint here to maintain consistency between props and defined values.

```suggestion
const sizes: Record<SelectProps['size'], string> = {
```